python/pyspark/ml/fpm.py

```diff
@@ -0,0 +1,232 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from pyspark import keyword_only, since
+from pyspark.ml.util import *
+from pyspark.ml.wrapper import JavaEstimator, JavaModel
+from pyspark.ml.param.shared import *
+
+__all__ = ["FPGrowth", "FPGrowthModel"]
+
+
+class HasSupport(Params):
+    """
+    Mixin for param support: [0.0, 1.0].
+    """
+
+    minSupport = Param(
+        Params._dummy(),
+        "minSupport",
+        "Minimal support level of the frequent pattern. [0.0, 1.0]. Any pattern that appears more "
+        "than (minSupport * size-of-the-dataset) times will be output",
+        typeConverter=TypeConverters.toFloat)
+
+    def setMinSupport(self, value):
+        """
+        Sets the value of :py:attr:`minSupport`.
+        """
+        if not (0 <= value <= 1):
```

Comment:
On this topic, I agree with you that not checking here could currently cause late failures in a Pipeline.  However, I think the right fix for this is to add PipelineStage and transformSchema() to Python.  I just made a JIRA for it: https://issues.apache.org/jira/browse/SPARK-20099